### ISSUE TITLE
GL Error when running Offscreen.py

### ISSUE THREAD

**Issue by ardot** [vispy/vispy#867]
Hey all,

Thanks for the library! I'm trying to use gloo for offscreen rendering, but am getting an error when I run the offscreen.py example script. Paste from the error message is at:

https://gist.github.com/ardot/f300779d24102f2d15d4

Is this a KP? Am I configuring something wrong?
Thanks,
-Tim

**Comment by almarklein** [comment]
Can you turn on gl debugging: `gloo.gl.use_gl('gl debug')`? This should provide us with a clearer error message. Also a minimal example should help.

**Comment by rossant** [comment]
I can reproduce this. The `offscreen.py` example in demo/gloo is broken on master.

**Comment by Eric89GXL** [comment]
It works for me. I wonder if this is the ol' Intel GPU (#649) issue coming up again. @ardot what GPU are you running?

**Comment by ardot** [comment]
Eric - Seems very possible. I have a late 2013 Macbook pro with an Intel Iris Pro 1024 MB GPU. 

Almar - when I add in the debugging line I get "RuntimeError: Could not import gl target "gl":
No module named gl". It seems like 'gl debug' isn't a valid input here. Thoughts?

**Comment by rossant** [comment]
I can confirm that I only have this bug on my Intel GPU, not my NVIDIA one.

**Comment by Eric89GXL** [comment]
@ardot try `use_gl('desktop debug')`

@ardot @rossant if one of you could boil the example down to a minimal version, it might also help us figure out where the failure occurs. Or if you want to ship me your laptop I can have a look :)

**Comment by ardot** [comment]
@Eric89GXL 'desktop debug' gives the same error, except replace "no module named desktop" instead of "no module named gl".

**Comment by Eric89GXL** [comment]
Are you on latest master?

**Comment by ardot** [comment]
Yes, I believe so. I just did a 'git pull', uninstalled the current version with 'pip uninstall vispy" and re-installed with 'python setup.py install'. Did I do that correctly? Is there anything else I need to do to update?

**Comment by Eric89GXL** [comment]
Agh, I forgot we changed the names. Try `use_gl('gl2 debug')`.

**Comment by ardot** [comment]
Nice! Yeah that works. I am still getting the same error messages though:
"OpenGL got errors (Check before draw): GL_INVALID_FRAMEBUFFER_OPERATION"

**Comment by Eric89GXL** [comment]
Although some posts make it seem like the Intel driver might not work properly:

https://communities.intel.com/message/220401

I wonder if we're not doing something right with the framebuffer, see first answer here:

http://stackoverflow.com/questions/10533609/opengl-gl-quads-produces-error-0x506

@rossant @ardot any interest in seeing if you can see what e.g. the status from `glCheckFramebufferStatusEXT` is?

**Comment by almarklein** [comment]
>  I am still getting the same error messages though:

But the trace should be more precise now. In debug mode, we test for GL errors on each GL command rather than only once each draw.

**Comment by Eric89GXL** [comment]
@almarklein I think the error could be slightly more subtle, see stack overflow link above. We could be creating the framebuffer without error, but it could be in a non-useable INCOMPLETE state. We might need to change which type of color we use, for example.

**Comment by ardot** [comment]
@almarklein  Gotchya. Okay, you were right - the trace is different. Here is the new trace:
https://gist.github.com/ardot/fda493f68afbe45ee31b

**Comment by almarklein** [comment]
This shows that the error is detected right after calling `glClear`. Which is a bit weird, since you'd expect to see it after FBO attachement or something. @Eric89GXL is probably right in that this is some subtle bug...

**Comment by Eric89GXL** [comment]
I think with the issue I linked to above you don't get the error during attachment, but only once you try to do something to the attached framebuffer (but the attachment is "incomplete"). So that's consistent with that idea. @ardot or @rossant if one of you has time to dig into trying different attachment schemes, even just in raw `gl` and seeing what throws the error and what fixes it, that might be the best approach.

**Comment by stonebig** [comment]
for the maybe same issue on `one_cam_two_scenes.py`, I do:
````
import numpy as np

from vispy import app, scene, io, gloo

canvas = scene.SceneCanvas(keys='interactive')
canvas.size = 800, 600
canvas.show()

# Create two ViewBoxes, place side-by-side
vb1 = scene.widgets.ViewBox(border_color='yellow', parent=canvas.scene)
vb2 = scene.widgets.ViewBox(border_color='blue', parent=canvas.scene)
#
grid = canvas.central_widget.add_grid()
grid.padding = 6
grid.add_widget(vb1, 0, 0)
grid.add_widget(vb2, 0, 1)

# Create the image
im1 = io.load_crate().astype('float32') / 255
# Make gray, smooth, and take derivatives: edge enhancement
im2 = im1[:, :, 1]
im2 = (im2[1:-1, 1:-1] + im2[0:-2, 1:-1] + im2[2:, 1:-1] + 
       im2[1:-1, 0:-2] + im2[1:-1, 2:]) / 5
im2 = 0.5 + (np.abs(im2[0:-2, 1:-1] - im2[1:-1, 1:-1]) + 
             np.abs(im2[1:-1, 0:-2] - im2[1:-1, 1:-1]))

image1 = scene.visuals.Image(im1, parent=vb1.scene)
image2 = scene.visuals.Image(im2, parent=vb2.scene)

# Set 2D camera (PanZoomCamera, TurnTableCamera)
vb1.camera, vb2.camera = scene.PanZoomCamera(), scene.PanZoomCamera()
vb1.camera.aspect = vb2.camera.aspect = 1  # no auto-scale
vb1.camera.link(vb2.camera)

# Set the view bounds to show the entire image with some padding
#view.camera.rect = (-10, -10, image.size[0]+20, image.size[1]+20)
gloo.gl.use_gl('gl2 debug')
if __name__ == '__main__':
    app.run()
````

and gets:
````
WARNING: Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\notebooks\one_cam_two_scene.py", line 58, in <module>
    app.run()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\_default_app.py", line 61, in run
    return default_app.run()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\application.py", line 123, in run
    return self._backend._vispy_run()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\backends\_qt.py", line 217, in _vispy_run
    return app.exec_()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\backends\_qt.py", line 631, in paintGL
    self._vispy_canvas.events.draw(region=None)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 436, in __call__
    self._invoke_callback(cb, event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 452, in _invoke_callback
    self, cb_event=(cb, event))
  << caught exception here: >>
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\_gl2.py", line 53, in glBindFramebuffer
    nativefunc = glBindFramebuffer._native
AttributeError: 'function' object has no attribute '_native'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\gl2.py", line 59, in _get_gl_func
    func = getattr(_lib, name)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\ctypes\__init__.py", line 364, in __getattr__
    func = self.__getitem__(name)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\ctypes\__init__.py", line 369, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
AttributeError: function 'glBindFramebuffer' not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 448, in _invoke_callback
    cb(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 189, in on_draw
    self._draw_scene()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 232, in _draw_scene
    self.draw_visual(self.scene, viewport=viewport)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 282, in draw_visual
    visual.draw(scene_event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 50, in draw
    self.process_system(event, 'draw')
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 60, in process_system
    self._systems[system_name].process(event, self)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 28, in process
    _handle_exception(False, 'reminders', self, node=node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 23, in process
    node.draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\widgets\viewbox.py", line 288, in draw
    super(ViewBox, self).draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\widgets\widget.py", line 236, in draw
    self._visual.draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\mesh.py", line 332, in draw
    self._program.draw(self._mode)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\shaders\program.py", line 89, in draw
    Program.draw(self, *args, **kwargs)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\program.py", line 467, in draw
    canvas.context.flush_commands()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\context.py", line 163, in flush_commands
    self.shared.parser.parse([('CURRENT', 0)])
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 401, in parse
    self._parse(command)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 329, in _parse
    gl.glBindFramebuffer(gl.GL_FRAMEBUFFER, 0)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\_proxy.py", line 35, in glBindFramebuffer
    self("glBindFramebuffer", False, target, framebuffer)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\__init__.py", line 85, in __call__
    ret = func(*args)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\_gl2.py", line 55, in glBindFramebuffer
    nativefunc = glBindFramebuffer._native = _get_gl_func("glBindFramebuffer", None, (ctypes.c_uint, ctypes.c_uint,))
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\gl2.py", line 77, in _get_gl_func
    raise RuntimeError('Function %s not present in context.' % name)
RuntimeError: Function glBindFramebuffer not present in context.
ERROR: Invoking <bound method SceneCanvas.on_draw of <SceneCanvas (PyQt4) at 0x3a3cef0>> for DrawEvent
ERROR: Invoking <bound method SceneCanvas.on_draw of <SceneCanvas (PyQt4) at 0x3a3cef0>> repeat 2
ERROR: Invoking <bound method SceneCanvas.on_draw of <SceneCanvas (PyQt4) at 0x3a3cef0>> repeat 3
````

**Comment by stonebig** [comment]
on the same switchable-graphics PC, I switch to the ATI HD-5650, I get another error, but two images in the vispy canvas this time are drawed:
````
WARNING: Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\notebooks\one_cam_two_scene.py", line 58, in <module>
    app.run()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\_default_app.py", line 61, in run
    return default_app.run()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\application.py", line 123, in run
    return self._backend._vispy_run()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\backends\_qt.py", line 217, in _vispy_run
    return app.exec_()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\backends\_qt.py", line 631, in paintGL
    self._vispy_canvas.events.draw(region=None)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 436, in __call__
    self._invoke_callback(cb, event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 452, in _invoke_callback
    self, cb_event=(cb, event))
  << caught exception here: >>
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 448, in _invoke_callback
    cb(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 189, in on_draw
    self._draw_scene()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 232, in _draw_scene
    self.draw_visual(self.scene, viewport=viewport)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 282, in draw_visual
    visual.draw(scene_event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 50, in draw
    self.process_system(event, 'draw')
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 60, in process_system
    self._systems[system_name].process(event, self)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 28, in process
    _handle_exception(False, 'reminders', self, node=node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 23, in process
    node.draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\widgets\viewbox.py", line 288, in draw
    super(ViewBox, self).draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\widgets\widget.py", line 236, in draw
    self._visual.draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\mesh.py", line 332, in draw
    self._program.draw(self._mode)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\shaders\program.py", line 89, in draw
    Program.draw(self, *args, **kwargs)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\program.py", line 467, in draw
    canvas.context.flush_commands()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\context.py", line 164, in flush_commands
    self.glir.flush(self.shared.parser)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 158, in flush
    parser.parse(self._filter(self.clear(), parser))
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 401, in parse
    self._parse(command)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 363, in _parse
    ob.draw(*args)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 847, in draw
    gl.glDrawArrays(mode, first, count)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\_proxy.py", line 179, in glDrawArrays
    self("glDrawArrays", False, mode, first, count)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\__init__.py", line 85, in __call__
    ret = func(*args)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\_gl2.py", line 415, in glDrawArrays
    nativefunc(mode, first, count)
OSError: exception: access violation reading 0x0861D368
ERROR: Invoking <bound method SceneCanvas.on_draw of <SceneCanvas (PyQt4) at 0x3a2cd10>> for DrawEvent
````

**Comment by stonebig** [comment]
curioustly, the donut.py example worsk on intem graphics, and shows nothing (a white window) on the ATI graphics, with no error reported.

**Comment by Eric89GXL** [comment]
I just tested the code on an NVIDIA switchable graphics system (Windows 8.1) and it worked for both GPUs for me. I wonder if Intel and ATI have iffy OpenGL support across different drivers / cards. I have an old laptop with Intel HD 3000 graphics I think that I can try, but I'm not sure even that will allow me to reproduce the problem.

**Comment by Eric89GXL** [comment]
Oh wait, there's hope... I had just been trying the camera example, which works fine for me. I can reproduce the elusive invalid framebuffer bug over here. I'll try to dig into it to see if I can solve it.

**Comment by stonebig** [comment]
testing by hand on isocurve on my intel-inside:
- now, the butons are working interactively,
 - I have still some message in my spyder console, I don't know if it's important ?
````
WARNING: Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\notebooks\isocurve_for_trisurface.py", line 129, in <module>
    appQt.exec_()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\backends\_qt.py", line 631, in paintGL
    self._vispy_canvas.events.draw(region=None)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 436, in __call__
    self._invoke_callback(cb, event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 452, in _invoke_callback
    self, cb_event=(cb, event))
  << caught exception here: >>
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\_gl2.py", line 53, in glBindFramebuffer
    nativefunc = glBindFramebuffer._native
AttributeError: 'function' object has no attribute '_native'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\gl2.py", line 59, in _get_gl_func
    func = getattr(_lib, name)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\ctypes\__init__.py", line 364, in __getattr__
    func = self.__getitem__(name)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\ctypes\__init__.py", line 369, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
AttributeError: function 'glBindFramebuffer' not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\util\event.py", line 448, in _invoke_callback
    cb(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 189, in on_draw
    self._draw_scene()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 232, in _draw_scene
    self.draw_visual(self.scene, viewport=viewport)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\canvas.py", line 282, in draw_visual
    visual.draw(scene_event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 50, in draw
    self.process_system(event, 'draw')
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 60, in process_system
    self._systems[system_name].process(event, self)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 28, in process
    _handle_exception(False, 'reminders', self, node=node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 23, in process
    node.draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\widgets\viewbox.py", line 324, in draw
    self.scene.draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 50, in draw
    self.process_system(event, 'draw')
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\subscene.py", line 60, in process_system
    self._systems[system_name].process(event, self)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 37, in process
    self.process(event, sub_node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 28, in process
    _handle_exception(False, 'reminders', self, node=node)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\scene\systems.py", line 23, in process
    node.draw(event)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\isoline.py", line 218, in draw
    LineVisual.draw(self, transforms)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\line\line.py", line 275, in draw
    self._line_visual.draw(transforms)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\line\line.py", line 393, in draw
    self._program.draw('lines', self._connect_ibo)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\visuals\shaders\program.py", line 89, in draw
    Program.draw(self, *args, **kwargs)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\program.py", line 467, in draw
    canvas.context.flush_commands()
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\context.py", line 163, in flush_commands
    self.shared.parser.parse([('CURRENT', 0)])
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 401, in parse
    self._parse(command)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\glir.py", line 329, in _parse
    gl.glBindFramebuffer(gl.GL_FRAMEBUFFER, 0)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\_gl2.py", line 55, in glBindFramebuffer
    nativefunc = glBindFramebuffer._native = _get_gl_func("glBindFramebuffer", None, (ctypes.c_uint, ctypes.c_uint,))
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\gloo\gl\gl2.py", line 77, in _get_gl_func
    raise RuntimeError('Function %s not present in context.' % name)
RuntimeError: Function glBindFramebuffer not present in context.
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> for DrawEvent
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 2
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 3
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 4
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 5
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 6
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 7
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 8
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 9
ERROR: Invoking <bound method Canvas.on_draw of <Canvas (PyQt4) at 0x62d7a50>> repeat 10
````

**Comment by stonebig** [comment]
the offscreen.py complains about a `GLFW`library not found 
````
Traceback (most recent call last):
  File "D:\result_tests\winpython-3.4.3.3\notebooks\offscreen.py", line 32, in <module>
    app.use_app('glfw')
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\_default_app.py", line 46, in use_app
    default_app = Application(backend_name)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\application.py", line 49, in __init__
    self._use(backend_name)
  File "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\application.py", line 204, in _use
    raise RuntimeError(msg)
RuntimeError: Could not import backend "Glfw":
GLFW library not found

````

**Comment by campagnola** [comment]
@stonebig, it would be helpful to use a regular console (cmd or powershell) to make sure we aren't also debugging spyder issues at the same time (but problems with spyder could be dealt with in another issue).

**Comment by stonebig** [comment]
You're right, there could be an interaction.
If I launch via IDLE, I confirm I get the same errors for isocurve_for_trisurface.py

**Comment by stonebig** [comment]
ah, replaying this example in AMD hd5650 mode, I get :
- zero message in the console,
- 3 bars referencial in the middle of the sphere, that I don't have when under intel graphics.

Would it mean that the problem is in the drawing of these little 3 bars ?
